src/components/chat/tool-icon-component.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useContext } from "react";
+import { useTypeOfTileInDocumentOrCurriculum } from "../../hooks/use-stores";
+import { AppConfigContext } from "../../app-config-context";
+
+interface IProps {
+  documentKey?: string;
+  tileId?: string;
+}
+
+export const ToolIconComponent: React.FC<IProps> = ({documentKey, tileId}) => {
+
+  const tileType = useTypeOfTileInDocumentOrCurriculum(documentKey, tileId);
+  const { appIcons } = useContext(AppConfigContext);
+  
+  if (!documentKey || !tileId || !appIcons ) {
+    return null;
+  }
+
+  if (!tileType) {
+    return <div>{appIcons?.["icon-open-workspace"]}</div>;
+  }
+
+  const iconName = `icon-${tileType.toLowerCase()}-tool`;
+  const TheIcon = appIcons?.[iconName];
+  return <div><TheIcon/></div>;
+};
```

Comment:
Do you have VSCode set up to auto-remove trailing spaces on save?
https://stackoverflow.com/questions/30884131/remove-trailing-spaces-automatically-or-with-a-shortcut

We tend to do this so that we don't get whitespace changes happening in modules from commit to commit (it tends to clutter the diff).  I notice that if I save this module on my system, VSCode strips out some excess whitespace:
![image](https://user-images.githubusercontent.com/5126913/135919892-739c37c0-9f1b-4d6f-8343-0d0c76981878.png)

Reply:
Also realizing that I just naively assumed you are using VSCode.

Reply:
Updated my VSCode settings to enable this. THANK you!